test/ng/sceSpecs.js

```diff
@@ -525,6 +552,44 @@ describe('SCE', function() {
     ));
   });
 
+  describe('URL-context sanitization', function() {
+    it('should sanitize values that are not whitelisted', inject(function($sce) {
+      expect($sce.getTrustedMediaUrl('javascript:foo')).toEqual('unsafe:javascript:foo');
+      expect($sce.getTrustedUrl('javascript:foo')).toEqual('unsafe:javascript:foo');
+    }));
+
+    it('should not sanitize values that are whitelisted', inject(function($sce) {
+      expect($sce.getTrustedMediaUrl('http://example.com')).toEqual('http://example.com');
+      expect($sce.getTrustedUrl('http://example.com')).toEqual('http://example.com');
+    }));
+
+    it('should not sanitize trusted values', inject(function($sce) {
+      expect($sce.getTrustedMediaUrl($sce.trustAsMediaUrl('javascript:foo'))).toEqual('javascript:foo');
+      expect($sce.getTrustedMediaUrl($sce.trustAsUrl('javascript:foo'))).toEqual('javascript:foo');
+      expect($sce.getTrustedMediaUrl($sce.trustAsResourceUrl('javascript:foo'))).toEqual('javascript:foo');
+
+      expect($sce.getTrustedUrl($sce.trustAsMediaUrl('javascript:foo'))).toEqual('unsafe:javascript:foo');
+      expect($sce.getTrustedUrl($sce.trustAsUrl('javascript:foo'))).toEqual('javascript:foo');
+      expect($sce.getTrustedUrl($sce.trustAsResourceUrl('javascript:foo'))).toEqual('javascript:foo');
+    }));
+
+    it('should use the $$sanitizeUri', function() {
```

Comment:
Isn't this technically an implementation detail?

Reply:
Well I see it as an interface (albeit internal) to the service. Testing it this way means that we are less tied to the implementation of the `$$sanitizeUri` service.